agent/xds/clusters_test.go

```diff
@@ -275,6 +275,12 @@ func TestClustersFromSnapshot(t *testing.T) {
 				return proxycfg.TestConfigSnapshotDiscoveryChain(t, "failover-to-cluster-peer", nil, nil)
 			},
 		},
+		{
+			name: "connect-proxy-with-chain-and-redirect-to-cluster-peer",
+			create: func(t testinf.T) *proxycfg.ConfigSnapshot {
+				return proxycfg.TestConfigSnapshotDiscoveryChain(t, "redirect-to-cluster-peer", nil, nil)
+			},
+		},
```

Comment:
RB showed me that you can accomplish the same thing as adding this test case in 4 different places by just adding the test to [resources_test.go](https://github.com/hashicorp/consul/blob/main/agent/xds/resources_test.go#L197-L218).